src/Chapter3.hs

```diff
@@ -910,6 +994,19 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+instance Append Gold where
+    append :: Gold -> Gold ->  Gold
+    append (Gold g1) (Gold g2) = Gold (g1 + g2)
+
+instance Append [a] where
+  append :: [a] -> [a] -> [a]
+  append = (++)
+
+instance Append m => Append (Maybe m) where
+  append :: Maybe m -> Maybe m -> Maybe m
+  append (Just j1) (Just j2) = Just (append j1 j2)
+  append _ _ = Nothing
```

Comment:
While this is valid implementation and nothing in the task description says differently, in reality it would be more beneficial is this function returned `Just` when at least one argument is `Just`.

Reply:
Yep, that makes sense